quilter.py

```diff
@@ -0,0 +1,360 @@
+    zip_list = []
+    z_prog = 0
+    for dir_name, subdir_list, file_list in os.walk(dl_folder):
+        for fname in file_list:
+            if str.endswith(fname, ".zip"):
+                zip_list.append(os.path.join(dir_name, fname))
+
+    for z in zip_list:
+        z_prog += 1
+        print("Extracting {}, {} of {}".format(z, z_prog, str(len(zip_list))))
+        with zipfile.ZipFile(z) as zip_ref:
+            zip_ref.extractall(unzip_folder)
+
+    # copy out all .img to output dir
+    print("\nCopying extracted files...")
+    os.mkdir(extract_folder)
+    extract_list = []
+    extract_prog = 0
+    for dir_name, subdir_list, file_list in os.walk(unzip_folder):
+        for fname in file_list:
+            if str.endswith(fname, ext_list):
+                extract_list.append(os.path.join(dir_name, fname))
+
+    for f in extract_list:
+        extract_prog += 1
+        f_name = os.path.basename(f)
+        print("Copying {}, {} of {}".format(f_name, extract_prog, 
+                                            str(len(extract_list))))
+        shutil.copy2(f, extract_folder)
+
+    # If we've gotten this far, we can go ahead and delete the downloaded zips
+    delete = True
+
+    # shapefile merging
+    if merge and vector:
+        shp_list = []
+
+        # Assumes gdal installed via Conda; unknown if this works with 
+        # OSG4W, qgis, etc.
+        # Really, if you're doing GDAL in python, you should use conda. 
+        # Seriously.
+        gdal_path = Path(inspect.getfile(gdal))
+        if r'Program Files\ArcGIS' in str(gdal_path):
+            raise RuntimeError("The GDAL environment bundled with ArcGIS Pro does not provide the script necessary for merging shapefiles. Recommend installing latest GDAL from the conda-forge channel: conda install -c conda-forge gdal.")
+        else: 
+            merge_path = gdal_path.parents[3] / 'Scripts' / 'ogrmerge.py'
+
+        if not os.path.exists(merge_path):
+            raise FileNotFoundError("ogrmerge.py not found where expected. Did you install GDAL from conda-forge?")
+
+        shp_args = ['python', merge_path, '-o', vector_outpath]
+
+        for dir_name, subdir_list, file_list in os.walk(extract_folder):
+            for fname in file_list:
+                if str.endswith(fname, ".shp"):
+                    shp_list.append(os.path.join(dir_name, fname))
+
+        subprocess.run(shp_args)
+    
+    # raster merging
+    elif merge and raster:
+        print("\nCombining into one dataset...")
+
+        # Merge files by building VRT
+        vrt_list = []
+        vrt_path = os.path.join(tdir, final_name + str(os.getpid()) +
+                                ".vrt")
+        for dir_name, subdir_list, file_list in os.walk(extract_folder):
+            for fname in file_list:
+                if str.endswith(fname, raster_exts):
+                    vrt_list.append(os.path.join(dir_name, fname))
+
+        vrt_opts = gdal.BuildVRTOptions(resampleAlg='cubic')
```

Comment:
dont' be afraid to use more descriptive variables. I'm starting to get lost with `vrt` and `co` and `ds` and `dso` and `md`. It might save you a bit of typing now, but readability suffers.

Reply:
Habits carried over from prototyping. I'll make them more explicit.

Reply:
thanks! me and future you will thank you